src/sqs/backfill.ts

```diff
@@ -42,3 +55,25 @@ export const backfillQueueMessageHandler: MessageHandler<BackfillMessagePayload>
 		throw err;
 	}
 };
+
+const getRateRateLimitStatus = async (backfillData: BackfillMessagePayload, logger: Logger) => {
+	const { installationId, jiraHost } = backfillData;
+	const gitHubAppId = backfillData.gitHubAppConfig?.gitHubAppId;
+	const gitHubInstallationClient = await createInstallationClient(installationId, jiraHost, logger, gitHubAppId);
+
+	return await gitHubInstallationClient.getRateLimit();
+};
+
+const isRateLimitExceedingSoftLimit = async (rateLimitResponse: Octokit.RateLimitGetResponse, jiraHost) : Promise<boolean> => {
+	const threshold = await numberFlag(NumberFlags.PREEMPTIVE_RATE_LIMIT_THRESHOLD, 100, jiraHost);
+	const { core, graphql } = rateLimitResponse.resources;
+	const usedPercentCore = ((core.limit - core.remaining) / core.limit) * 100;
+	const usedPercentGraphql = ((graphql.limit - graphql.remaining) / graphql.limit) * 100;
+
+	return usedPercentCore >= threshold || usedPercentGraphql >= threshold;
```

Comment:
There is 5 or so rate limits, we care about core(REST) and graphql and if either of them are running close then we delay, they are both 5000 from testing my cloud instance.

Incase of interest.... the other three rate limits are search, code_scanning_upload and integration_manifest.

There is also a deprecated rate object that we should not use!